### PR TITLE
Add dockerization for easy deployment

### PR DESCRIPTION
**Pull request by schemen**
Also, I've changed the way configs are loaded. Also added is that we can change the savefile location.

### CODE REVIEW

**Comment by Kruptein** [comment]
Hey, can you do the pr on the dev branch instead? I follow gitflow which uses the master is a release branch.

Also ignore the appveyor build, i have yet to set it up

**Comment by schemen** [comment]
Hey, I just changed the into branch to dev :)  you can use this PR still :)

**Comment by Kruptein** [comment]
Oh, silly me, i'll look at the pr more in depth tonight

**Comment by schemen** [comment]
Cheers! Let me know if you want anything changed. This branch is currently deployed to my cluster at https://planarally.dndbox.com

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+WORKDIR /usr/src/app
+
+VOLUME /usr/src/app/data
+VOLUME /usr/src/app/static/assets
+
+ENTRYPOINT ["/usr/bin/dumb-init", "--"]
+
+COPY requirements.txt ./
+RUN apt-get update && apt-get install dumb-init && \
```
Comment (FrederikGoovaerts) [inline Dockerfile]: I think it's a good idea to do `rm -rf /var/lib/apt/lists/*` after the install completed. The docker container won't need the apt list anymore and this takes up a decent amount of space.

File: Dockerfiles/server_config.cfg [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Webserver]
```
Comment (FrederikGoovaerts) [inline Dockerfiles/server_config.cfg]: Why is this file duplicated now? If I interpret correctly, you copy everything in the Planarally folder, including the cfg file there, then copy this one over the original one. Am I missing something?

File: PlanarAlly/planarserver.py [hunk 3]
```diff
@@ -871,9 +875,7 @@ def shape_wrap(player, shape):
 app.on_shutdown.append(on_shutdown)
 
 if __name__ == '__main__':
-    app['background_save'] = sio.start_background_task(save_all)
-    cfg = configparser.ConfigParser()
-    cfg.read("server_config.cfg")
+    sio.start_background_task(save_all)
```
Comment (Kruptein) [inline PlanarAlly/planarserver.py]: You want to keep the app['background_save'] assignment, as it is used on line 84 to cancel the task.

Comment (schemen) [reply PlanarAlly/planarserver.py]: Done that!

File: Dockerfile (hunk 1 above)
Comment (schemen) [reply Dockerfile]: Good Idea, done that.

File: Dockerfiles/server_config.cfg (hunk 2 above)
Comment (schemen) [reply Dockerfiles/server_config.cfg]: I'm only copying the content of the folder PlanarAlly/, not the entire thing.

File: PlanarAlly/planarserver.py (hunk 3 above)
Comment (Kruptein) [reply PlanarAlly/planarserver.py]: Now you start the background save twice ;)

Comment (schemen) [reply PlanarAlly/planarserver.py]: I have updated it, do you mean like that?

Comment (Kruptein) [reply PlanarAlly/planarserver.py]: indeed

**Comment by schemen** [comment]
Ok, so far all remarks are resolved. Anything else I could improve :) ?

**Pull request merged**